controllers/infinispan_secret_controller.go

```diff
@@ -109,33 +116,160 @@ func (reconciler *SecretReconciler) Reconcile(ctx context.Context, request recon
+			var ret string
+			for i, bs := range elems {
+				ret += fmt.Sprintf("%s[%d]", bs.Address, bs.Port)
+				if i < len(elems)-1 {
+					ret += ","
+				}
+			}
+			return ret
+		},
+	}
+	var ispnXmlTemplate, jgroupsXmlTemplate string
+	if box, err := rice.FindBox("resources/"); err != nil {
+		return &reconcile.Result{}, err
+	} else {
+		if ispnXmlTemplate, err = box.String("ispnXmlTemplate.xmltmpl"); err != nil {
+			return &reconcile.Result{}, err
+		}
+		if jgroupsXmlTemplate, err = box.String("jgroupsXmlTemplate.xmltmpl"); err != nil {
+			return &reconcile.Result{}, err
+		}
+	}
+
+	// Generate infinispan.xml
+	name := r.infinispan.Name
+	namespace := r.infinispan.Namespace
+	infinispanXmlObject := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.infinispan.GetInfinispanXmlConfigName(),
+			Namespace: namespace,
+		},
+	}
+	tIspn, err := template.New("infinispan.xml").Funcs(funcMap).Parse(ispnXmlTemplate)
 	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	buffIspn := new(bytes.Buffer)
+	err = tIspn.Execute(buffIspn, serverConf)
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+
+	var buffJGroups *bytes.Buffer
+	if serverConf.XSite != nil && len(serverConf.XSite.Backups) > 0 {
+		// Generate jgroups-relay.xml
+		tJGroups, err := template.New("jgroups-relay.xml").Funcs(funcMap).Parse(jgroupsXmlTemplate)
+		if err != nil {
+			return &reconcile.Result{}, err
+		}
+		buffJGroups = new(bytes.Buffer)
+		err = tJGroups.Execute(buffJGroups, serverConf)
+		if err != nil {
+			return &reconcile.Result{}, err
+		}
+	}
+
+	// Create user, admin identity properties from secrets
+	var users, groups string
+	if userPropSecret != nil {
+		users, groups, err = security.AuthPropsFromSecret(userPropSecret.Data[consts.ServerIdentitiesFilename])
+		if err != nil {
+			return &reconcile.Result{}, err
+		}
+	}
+	adminUsers, adminGroups, err := security.AuthPropsFromSecret(adminPropSecret.Data[consts.ServerIdentitiesFilename])
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+
+	// Create secret with all the objects to be mounted as "ServerRoot/conf/operator/"
+	result, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, infinispanXmlObject, func() error {
+		infinispanXmlObject.Labels = LabelsResource(r.infinispan.Name, "infinispan-secret-admin-identities")
+		infinispanXmlObject.Data = map[string][]byte{"infinispan.xml": buffIspn.Bytes()}
+		if buffJGroups != nil {
+			infinispanXmlObject.Data["jgroups-relay.xml"] = buffJGroups.Bytes()
+		}
+		infinispanXmlObject.Data[consts.ServerUsersPropertiesFilename] = []byte(users)
+		infinispanXmlObject.Data[consts.ServerGroupsPropertiesFilename] = []byte(groups)
+		infinispanXmlObject.Data[consts.ServerAdminUsersPropertiesFilename] = []byte(adminUsers)
+		infinispanXmlObject.Data[consts.ServerAdminGroupsPropertiesFilename] = []byte(adminGroups)
+		err = controllerutil.SetControllerReference(r.infinispan, infinispanXmlObject, r.scheme)
```

Comment:
Admittedly I'm biased as it was my idea, but I do think utilising the `IDENTITIES_BATCH` env var that will now be provided in the image will be much simpler than this. https://github.com/infinispan/infinispan-images/pull/154.

We can still allow the user to configure identities the same way as before and then instead of `AuthPropsFromSecret` we can have something like:

```go
func IdentitiesBatchFromSecret(buf []byte, usersFile, groupsFile string) (string, error) {
	var creds IdentitiesYaml
	if err := yaml.Unmarshal(buf, &creds); err != nil {
		return "", err
	}
	var b strings.Builder
	for _, cred := range creds.Credentials {
		fmt.Fprintf(&b, "user create %s -p %s --groups %s --users-file %s --groups-file %s",
			cred.Username, cred.Password, strings.Join(cred.Roles, ","), usersFile, groupsFile)
	}
	return b.String(), nil
}
```
This means that only the file generated by ^  needs to be mounted in the Secret.

Reply:
> * provide a reasonable set of scrambling methods for password

Here's what the CLI provides out of the box for `user create ryan -p password`:

```
ryan=scram-sha-1\:BYGcIAyiLM2/Wxh8vI6WA+ScCCl9RcM8c8oMTS6VFoU5Cx6oxg\=\=;scram-sha-256\:BYGcIAwrVv9Y5ULVKI+Uuu7S6yPtD6+90Df1tyGBTPWGFzQ1oHXURQLpYLsT4L2KvQ\=\=;scram-sha-384\:BYGcIAzKAzI2UmhG7ubUP2lvNrc00nQ6EOCKqrqAiuvU/0VBI5qD5pc+Qrr4aiYRB4f3+1KQxxwec8Z8oDDlbNI\=;scram-sha-512\:BYGcIAzXYdeDBfX3kEbB+jEGwYIJV0d4utZOwREKFpb55qzXEWYpxvTwrsfHYS+i5/0afrK0DDGV9wVi0A5zpjD89IgCKMpfUMsRvPgme7yW;digest-md5\:AgRyeWFuB2RlZmF1bHSDh6kdjRUvRH8RcRWLc2wM;digest-sha\:AgRyeWFuB2RlZmF1bHQwZDTkq9yq52mpCs0Jl5Ql2wvcgA\=\=;digest-sha-256\:AgRyeWFuB2RlZmF1bHSo/ZiOyoDoHV/1hN1XX5fq/YgLc1LqwzJt3codqO3jtQ\=\=;digest-sha-384\:AgRyeWFuB2RlZmF1bHSrLBC426STY96x0pCzOk5lTpcuqMtUeZpBK/XsJsYmn9SmPbC2uwfNF3t32D7voL4\=;digest-sha-512\:AgRyeWFuB2RlZmF1bHQUjxSw5d+KY1+5eTiYdW9lCwaPlC0jNChS6Hik+8HRY07QAvQZRWpLjNRXaVx2Kn+8pz2NpoGQ+EGDxRCuNOtK;
```